source/crud/crud.rst

```diff
@@ -336,165 +338,209 @@ Read
      *
-     * @see http://docs.mongodb.org/manual/reference/method/cursor.skip/
+     * @see https://docs.mongodb.com/manual/reference/command/find/
      */
-    skip: Optional<Int32>;
+    showRecordId: Optional<Document>;
 
     /**
-     * The order in which to return matching documents. If $orderby also exists
-     * in the modifiers document, the sort field overwrites $orderby.
-     * The default is no sort.
+     * The number of documents to skip before returning.
      *
-     * @see http://docs.mongodb.org/manual/reference/method/cursor.sort/
-     */ 
-    sort: Optional<Document>;
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     * For servers < 3.2, this is a wire protocol parameter that defaults to 0.
+     *
+     * @see https://docs.mongodb.com/manual/reference/command/find/
+     */
+    skip: Optional<Int64>;
 
-    /**
-     * Optionally specifies a collation to use in MongoDB 3.4 and higher.
+    /** 
+     * Determines whether to return the record identifier for each document. If true, adds a field $recordId to the returned documents.
      *
-     * The default is no value: the driver sends no "collation" document to the
-     * server with the "find" command.
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
      *
-     * @see http://docs.mongodb.org/manual/reference/command/find/
+     * @see https://docs.mongodb.com/manual/reference/command/find/
      */
-    collation: Optional<Document>;
+    snapshot: Optional<Document>;
 
+    /**
+     * The order in which to return matching documents.
+     *
+     * This option is sent only if the caller explicitly provides a value. The default is to not send a value.
+     *
+     * @see https://docs.mongodb.com/manual/reference/command/find/
+     */ 
+    sort: Optional<Document>;
   }
 
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Combining Limit and Batch Size for the Wire Protocol
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The OP_QUERY wire protocol only contains a numberToReturn value which drivers must calculate to get expected limit and batch size behavior. Subsequent calls to OP_GETMORE should use the user-specified batchSize or default to 0. Below is pseudo-code for calculating numberToReturn for OP_QUERY.
+The OP_QUERY wire protocol only contains a numberToReturn value which drivers must calculate to get expected limit and batch size behavior. Subsequent calls to OP_GETMORE should use the user-specified batchSize or default to 0. If the result is larger than the max Int32 value, an error MUST be raised as the computed value is impossible to send to the server. Below is pseudo-code for calculating numberToReturn for OP_QUERY.
 
 .. code:: typescript
 
-  function calculateFirstNumberToReturn(FindOptions options) {
+  function calculateFirstNumberToReturn(FindOptions options) Int32 {
```

Comment:
depends on what language we are using :). I changed this and the argument as well, to typescript-ish